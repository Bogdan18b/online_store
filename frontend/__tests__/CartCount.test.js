import CartCount from '../components/CartCount';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: "asfasgdfvfrewgf",
  title: 'shoes',
  price: 60000,
  description: 'really cool snickers',
  image: 'shoe.jpg',
  largeImage: 'shoe.jpg',
};

describe('<CartCount />', () => {
  it('renders', () => {
    shallow(<CartCount count={22} />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<CartCount count={22} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('updates via props', () => {
    const wrapper = shallow(<CartCount count={22} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});
