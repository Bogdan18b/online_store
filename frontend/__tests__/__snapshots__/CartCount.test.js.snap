// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> matches the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="22"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        22
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> updates via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="22"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        22
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> updates via props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="shoes"
    src="shoe.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "asfasgdfvfrewgf",
          },
        }
      }
    >
      <a>
        shoes
      </a>
    </Link>
  </Title>
  <PriceTag>
    $600
  </PriceTag>
  <p>
    really cool snickers
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "asfasgdfvfrewgf",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="asfasgdfvfrewgf"
    />
    <DeleteItem
      id="asfasgdfvfrewgf"
    />
  </div>
</ItemStyles__Item>
`;
